unime/src/routes/+layout.svelte

```diff
@@ -57,46 +59,121 @@
       goto(`/prompt/${type}`);
     }
   }
+
+  interface DevModeButton {
+    stringIcon?: string;
+    svelteIcon?: typeof SvelteComponent<SvelteHTMLElements['svg']>;
+    onClick: () => void;
+  }
+
+  function createDevButtons(): DevModeButton[] {
+    const backBtn: DevModeButton = {
+      svelteIcon: ArrowLeft,
+      onClick: () => history.back(),
+    };
+
+    const resetBtn: DevModeButton = {
+      svelteIcon: Trash,
+      onClick: () => dispatch({ type: '[App] Reset' }),
+    };
+
+    const ferrisBtn: DevModeButton = {
+      stringIcon: '🦀',
+      onClick: () => loadFerrisProfile(),
+    };
+
+    const dragonBtn: DevModeButton = {
+      stringIcon: '🐲',
+      onClick: () => (showDragonProfileSteps = !showDragonProfileSteps),
+    };
+
+    const debugBtn: DevModeButton = {
+      svelteIcon: ScrollText,
+      onClick: () => (showDebugMessages = !showDebugMessages),
+    };
+
+    return [backBtn, resetBtn, debugBtn, ferrisBtn, dragonBtn];
+  }
+
+  const devButtons = createDevButtons();
+
+  // Order needs to match the BE: 'ProfileSteps' enum, it needs to be the same order because every step is based upon the previous.
+  // 'AddCredentials' is ran after 'CreateProfile' and 'AcceptCredentials' after 'AddCredentials', etc.
+  const profileSteps: ProfileSteps[] = [
+    'CreateProfile',
+    'AddCredentials',
+    'AcceptCredentials',
+    'AddConnection',
+    'AcceptConnection',
+    'AddPresentation',
+    'ShareCredentails',
+    'AddFutureEngineer',
+    'CompleteFlow',
+  ];
+
+  async function loadFerrisProfile() {
+    dispatch({ type: '[DEV] Load DEV profile', payload: { profile: 'Ferris', execute_steps: null } }).then(() => {
+      // Reload page
+      setTimeout(async () => {
+        await goto('/');
+        await goto('/me');
+      }, 500);
+    });
+  }
+
+  async function loadDragonProfile(steps: ProfileSteps) {
+    await dispatch({
+      type: '[DEV] Load DEV profile',
+      payload: {
+        profile: 'Dragon',
+        execute_steps: steps,
+      },
+    });
+
+    showDragonProfileSteps = false;
+
+    if (steps == 'CompleteFlow') {
+      // Reload page
+      setTimeout(async () => {
+        await goto('/');
+        await goto('/me');
+      }, 500);
+    }
+  }
 </script>
 
 <main class="absolute h-screen">
   <!-- Dev Mode: Navbar -->
-  {#if $state?.dev_mode_enabled}
-    {#if showDevMode}
+  {#if $state?.dev_mode !== 'Off'}
+    {#if expandDevMenu}
       <div
-        class="hide-scrollbar fixed z-20 flex w-full space-x-4 overflow-x-auto bg-gradient-to-r from-red-200 to-red-300 p-4 shadow-md"
+        class="hide-scrollbar fixed z-20 flex w-full content-center overflow-x-auto bg-gradient-to-r from-red-200 to-red-300 p-4 pt-8 shadow-md"
```

Comment:
Why is there a `pt-8` added? Having the dev buttons closer to the top gives the content more room. Assuming you added it to not overlay the little arrow ("expand menu button") and make the dev buttons better reachable, I don't really see why the buttons themselves got smaller and less convenient now?